src/stores/maintenanceTaskStore.ts

```diff
@@ -0,0 +1,134 @@
+// @ts-strict-ignore
+import type { InstallValidation } from '@comfyorg/comfyui-electron-types'
+import { defineStore } from 'pinia'
+import { computed, ref } from 'vue'
+
+import { DESKTOP_MAINTENANCE_TASKS } from '@/constants/desktopMaintenanceTasks'
+import type {
+  MaintenanceTask,
+  MaintenanceTaskState
+} from '@/types/desktop/maintenanceTypes'
+import { electronAPI } from '@/utils/envUtil'
+import { useMinLoadingDurationRef } from '@/utils/refUtil'
+
+export const useMaintenanceTaskStore = defineStore('maintenanceTask', () => {
+  /** Refresh should run for at least this long, even if it completes much faster. Ensures refresh feels like it is doing something. */
+  const minRefreshTime = 250
+  const electron = electronAPI()
+
+  const isRefreshing = useMinLoadingDurationRef(false, minRefreshTime)
+  const isRunningTerminalCommand = computed(() =>
+    tasks.value
+      .filter((task) => task.usesTerminal)
+      .some((task) => getState(task)?.executing)
+  )
+  const isRunningInstallationFix = computed(() =>
+    tasks.value
+      .filter((task) => task.isInstallationFix)
+      .some((task) => getState(task)?.executing)
+  )
+
+  const taskStateMap = new Map<string, MaintenanceTaskState>()
```

Comment:
`ref<new Map<string, MaintenanceTaskState>>()` if here you intend to make MaintenanceTask reactive.